### PR TITLE
fix: Issues 60 and 134

### PR DESCRIPTION
**Pull request by Marwes**

### CODE REVIEW

File: base/src/types.rs [hunk 1]
```diff
@@ -800,6 +800,14 @@ impl<'a, I, T, E> DisplayType<'a, I, T, E>
     fn pretty(&self, arena: &'a Arena<'a>) -> DocBuilder<'a, Arena<'a>>
         where I: AsRef<str>,
     {
+        fn ident<'b>(arena: &'b Arena<'b>, name: &'b str) -> DocBuilder<'b, Arena<'b>> {
+            if name.starts_with(is_operator_char) {
```
Comment (brendanzab) [inline base/src/types.rs]: Would prefer `ast::is_operator_char` here

File: check/src/typecheck.rs [hunk 2]
```diff
@@ -42,6 +43,8 @@ pub enum TypeError<I> {
     Rename(::rename::RenameError),
     /// Multiple types were declared with the same name in the same expression
     DuplicateTypeDefinition(I),
+    /// A field were defined more than once in a record constructor or pattern match
```
Comment (brendanzab) [inline check/src/typecheck.rs]: "A field that was defined"

File: check/src/typecheck.rs [hunk 3]
```diff
@@ -642,46 +649,53 @@ impl<'a> Typecheck<'a> {
...
+                        });
+                    }
+                }
+
+                let mut new_fields: Vec<Field<_, _>> = Vec::with_capacity(fields.len());
+                for field in fields {
+                    let typ = try!(match field.1 {
+                        Some(ref mut expr) => Ok(self.typecheck(expr)),
+                        None => self.find(&field.0),
+                    });
```
Comment (brendanzab) [inline check/src/typecheck.rs]: Would this work?

```rust
let typ = match field.1 {
    Some(ref mut expr) => self.typecheck(expr),
    None => try!(self.find(&field.0)),
};
```

File: check/src/typecheck.rs (hunk 2 above)
Comment (Marwes) [inline check/src/typecheck.rs]: `"A field that was defined more than once in a record constructor or pattern match" does not seem quite right?

Comment (brendanzab) [inline check/src/typecheck.rs]: Oh woops: "A field was defined more than once in a record constructor or pattern match"

Comment (Marwes) [inline check/src/typecheck.rs]: Aha, 'were' vs' was. damn you proper English

**Pull request merged**